### PR TITLE
Pic 4372

### PR DESCRIPTION
**Pull request by jpanesar07**

### CODE REVIEW

File: helm_deploy/court-case-service/templates/ingress.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ metadata:
     # Secure the retry all dlqs endpoint from outside of the Kubernetes ingress
     nginx.ingress.kubernetes.io/server-snippet: |
       server_tokens off;
-      location ~* ^/(queue-admin/retry-all-dlqs|process-un-resulted-cases) {
+      location ~* ^/(queue-admin/retry-all-dlqs|process-un-resulted-cases|hearing/delete-duplicates) {
```
Comment (markreesmoj) [inline helm_deploy/court-case-service/templates/ingress.yaml:24]: strange that this does not have a leading slash but as long as it works then it is fine

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -201,6 +201,7 @@ offender-service:
 feature:
   flags:
     enable-cacheable-case-list: true
+    delete-hearing: true
```
Comment (markreesmoj) [inline src/main/resources/application.yml:204]: Don't we just want to set this to `true` in preprod and prod when reqjuired? I don't see the point in defining it here if we are overriding it in all environments

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/ImmutableCourtCaseService.java [hunk 3]
```diff
@@ -162,7 +162,7 @@ public List<HearingEntity> filterHearings(HearingSearchFilter hearingSearchFilte
     @Override
     public Optional<CourtCaseEntity> findByCaseId(String caseId) {
         assert courtCaseRepository != null;
```
Comment (markreesmoj) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/service/ImmutableCourtCaseService.java:164]: !!

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/DeleteHearingsService.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+@Slf4j
+@Service
+public class DeleteHearingsService {
+
+    private final HearingRepositoryFacade hearingRepositoryFacade;
+    private final DuplicateHearingRepository duplicateHearingRepository;
+    private final TelemetryService telemetryService;
+
+    @Autowired
+    private FeatureFlags featureFlags;
```
Comment (markreesmoj) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/service/DeleteHearingsService.java:24]: why isn't this in the constructor?

File: src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingRepository.java [hunk 5]
```diff
@@ -91,7 +91,8 @@ List<HearingEntity> findByCourtCodeAndHearingDay(
     @Query(value = "select * from hearing where id in (" +
         "select max(h.id) from hearing h, court_case cc where " +
         "cc.case_id = :caseId " +
-        "and h.fk_court_case_id = cc.id group by h.hearing_id)",
+        "and h.deleted = false " +
```
Comment (markreesmoj) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingRepository.java:94]: do we need to add this line to all these native queries? 😱

File: src/main/java/uk/gov/justice/probation/courtcaseservice/service/DeleteHearingsService.java (hunk 4 above)
Comment (markreesmoj) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/service/DeleteHearingsService.java:24]: I can see now - it is being modified in the IntegrationTest post construction

File: src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingRepository.java (hunk 5 above)
Comment (jpanesar07) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/jpa/repository/HearingRepository.java:94]: I know, only that one so far.

**Review by markreesmoj — APPROVED** [review]
I will have a proper look later on but for now let's give this a try in non-production environments

**Review by markreesmoj — COMMENTED** [review]
💬

File: src/main/java/uk/gov/justice/probation/courtcaseservice/application/FeatureFlags.java [hunk 6]
```diff
@@ -2,9 +2,12 @@
 
 import java.util.HashMap;
 import java.util.Map;
+
+import lombok.Getter;
 import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.stereotype.Component;
 
+@Getter
```
Comment (markreesmoj) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/application/FeatureFlags.java:10]: why is this necessary?
